### PR TITLE
Add OCP cluster upgrade feature

### LINKED ISSUE #38: Add support for cluster upgrade
Support running of upgrade playbook via terraform

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Fixes:  #38 
Signed-off-by: Prajyot Parab <prajyot.parab@ibm.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,6 +98,9 @@ Update the var.tfvars file with values specific to your environment. Following i
  * `storage_type` : Storage provisioner to configure. Supported values: nfs (For now only nfs provisioner is supported, any other value won't setup a storageclass)
  * `volume_size` : If storage_type is nfs, a volume will be created with given size in GB and attached to bastion node. Eg: 1000 for 1TB disk.
  * `volume_storage_template` : Storage template name or ID for creating the volume. Empty value will use default template.
+ * `upgrade_image` : If set, OCP cluster will be upgraded to this image version.
```
Comment (yussufsh) [inline README.md]: May be a little more info here. That it is actually "openshift release image". Also, the upgrade image version should be greate than the current version and should be supported.

File: variables.tf [hunk 2]
```diff
@@ -241,3 +241,18 @@ variable "volume_storage_template" {
     default = ""
 }
 
+variable "upgrade_image" {
+    #OCP upgrade image
+    default = ""
+}
+
+variable "pause_time" {
```
Comment (yussufsh) [inline variables.tf]: Since pause_time & delay_time is specific to upgrade.. rename them to upgrade_{pause,delay}_time.

File: README.md [hunk 3]
```diff
@@ -98,6 +98,9 @@ Update the var.tfvars file with values specific to your environment. Following i
...
+ * `pause_time` : Pauses playbook execution for a set amount of time in minutes.
```
Comment (yussufsh) [inline README.md]: The description does not make sense as a user.
pause_time: Minutes to wait before starting to check the upgrade status once the upgrade command is run?
delay_time: Seconds to wait before re-checking the upgrade status subsequently?

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (bpradipt) [reply README.md]: Also an example command line will be very helpful for users.

**Review by yussufsh — APPROVED** [review]

**Pull request merged**